test/PushNotificationTests/BaseTestSuite.cpp

```diff
@@ -0,0 +1,264 @@
+        VERIFY_SUCCEEDED(ChannelRequestHelper(channelOperation));
+    }
+    else
+    {
+        auto channelOperation{ PushNotificationManager::Default().CreateChannelAsync(c_azureRemoteId) };
+        VERIFY_ARE_EQUAL(ChannelRequestHelper(channelOperation), E_FAIL);
+    }
+}
+
+// Currently failing - https://github.com/microsoft/WindowsAppSDK/issues/2392
+void BaseTestSuite::MultipleChannelClose()
+{
+    auto channelOperation{ PushNotificationManager::Default().CreateChannelAsync(c_azureRemoteId) };
+    if (channelOperation.wait_for(c_timeout) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        channelOperation.Cancel();
+        VERIFY_FAIL(L"Channel request hit timeout.");
+    }
+
+    auto result{ channelOperation.GetResults() };
+    VERIFY_ARE_EQUAL(result.Status(), PushNotificationChannelStatus::CompletedSuccess);
+
+    result.Channel().Close();
+    VERIFY_THROWS_HR(result.Channel().Close(), WPN_E_CHANNEL_CLOSED);
+}
+
+void BaseTestSuite::VerifyRegisterAndUnregister()
+{
+    PushNotificationManager::Default().Register();
+    m_registered = true;
+
+    PushNotificationManager::Default().Unregister();
+}
+
+void BaseTestSuite::VerifyUnregisterFails()
+{
+    if (PushNotificationManager::Default().IsSupported())
+    {
+        VERIFY_THROWS_HR(PushNotificationManager::Default().Unregister(), E_UNEXPECTED);
+    }
+    else
+    {
+        PushNotificationManager::Default().Unregister();
+    }
+}
+
+void BaseTestSuite::VerifyUnregisterAllFails()
+{
+    if (PushNotificationManager::Default().IsSupported())
+    {
+        VERIFY_THROWS_HR(PushNotificationManager::Default().UnregisterAll(), E_UNEXPECTED);
+    }
+    else
+    {
+        PushNotificationManager::Default().UnregisterAll();
+    }
+}
+
+void BaseTestSuite::VerifyRegisterAndUnregisterAll()
+{
+    PushNotificationManager::Default().Register();
+    m_registered = true;
+
+    PushNotificationManager::Default().UnregisterAll();
+    m_registered = false;
+}
+
+void BaseTestSuite::MultipleRegister()
+{
+    PushNotificationManager::Default().Register();
+    m_registered = true;
+
+    if (PushNotificationManager::Default().IsSupported())
+    {
+        VERIFY_THROWS_HR(PushNotificationManager::Default().Register(), HRESULT_FROM_WIN32(ERROR_ALREADY_REGISTERED));
+    }
+    else
+    {
+        PushNotificationManager::Default().Register();
+    }
+}
+
+void BaseTestSuite::VerifyMultipleRegisterAndUnregister()
+{
+    PushNotificationManager::Default().Register();
+    m_registered = true;
+
+    PushNotificationManager::Default().Unregister();
+
+    if (PushNotificationManager::Default().IsSupported())
+    {
+        VERIFY_THROWS_HR(PushNotificationManager::Default().Register(), HRESULT_FROM_WIN32(ERROR_ALREADY_REGISTERED));
+    }
+    else
+    {
+        PushNotificationManager::Default().Register();
+    }
+}
+
+void BaseTestSuite::VerifyMultipleRegisterAndUnregisterAll()
+{
+    PushNotificationManager::Default().Register();
```

Comment:
The pattern:
 PushNotificationManager::Default().Register();
m_registered = true;
is repeated all over the place. So makes sense to have a helper called:
RegisterWithPushNotificationManager()